src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,97 @@
+package frc.robot.subsystems.shooter;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.math.Matrix;
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.controller.LinearQuadraticRegulator;
+import edu.wpi.first.math.estimator.KalmanFilter;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.system.LinearSystem;
+import edu.wpi.first.math.system.LinearSystemLoop;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.UnitModel;
+import frc.robot.utils.Utils;
+import org.apache.commons.lang.NullArgumentException;
+
+import static frc.robot.Constants.LOOP_PERIOD;
+import static frc.robot.Constants.NOMINAL_VOLTAGE;
+import static frc.robot.Constants.Shooter.*;
+import static frc.robot.Ports.Shooter.*;
+
+public class Shooter extends SubsystemBase {
+    private static final Shooter INSTANCE = new Shooter();
```

Comment:
The instance should be lazy initialized

Reply:
I'm not sure actually, what advantage does it give?
@barel13